### ISSUE TITLE
NeuroML v2 versions of channels?

### ISSUE THREAD

**Issue by JustasB** [OpenSourceBrain/PurkinjeCell#2]
Are there NML v2 versions of the channels somewhere? I could not find them in this project, and the converter resulted in files with "Not yet implemented" message. Are there any plans to support such conversion somewhere?

Details below:
[generatedNeuroML2](https://github.com/OpenSourceBrain/PurkinjeCell/tree/master/neuroConstruct/generatedNeuroML2) folder only has the leak channel.

Using the [ChannelMLConvert](https://sourceforge.net/p/neuroml/code/HEAD/tree/NeuroML2/ChannelMLConvert/) converter on e.g. [this .xml channel file](https://github.com/OpenSourceBrain/PurkinjeCell/blob/master/neuroConstruct/cellMechanisms/NaP/NaP_Chan.xml) (or other channel files), results in an .nml file with "Not implemented yet" instead of gating dynamics.

It seems that the converter has problems with the [current_voltage_relation construct seen on this line](https://github.com/OpenSourceBrain/PurkinjeCell/blob/master/neuroConstruct/cellMechanisms/NaP/NaP_Chan.xml#L71).

CC: @scrook

**Comment by JustasB** [comment]
@pgleeson Can you let us know if there are NMLv2 versions of the channels available for this model? Or if there are any ongoing v1->v2 conversion plans, or updates to the v1->v2 converter?

CC: @scrook

**Comment by pgleeson** [comment]
@JustasB Sorry I missed this the first time round. There is not a NMLv2 version of this model since the NeuroMLv1 version was never fully validated as matching the original GENESIS code. And even then, there is an issue about the stability/correctness of the GENESIS model which had been converted to Neuron and highlighted some issues: https://github.com/OpenSourceBrain/PurkinjeCell/issues/1.

So this project basically needs someone to spend time going through the original paper and code, and understanding the issues and converting what's useful to convert. I've asked a few students over the years and no joy yet...

A more likely scenario is to find someone to convert a more recent Purkinje cell model, e.g. https://senselab.med.yale.edu/ModelDB/ShowModel.cshtml?model=229585, but that's not started yet.

So in short, I wouldn't reuse any components of this model in NML2 in NeuroML-DB. As mentioned to @scrook, the single star for NeuroML2 support here: http://www.opensourcebrain.org/projects/purkinjecell should be a red flag. Please do ask though if it does look like a project/model is there in valid NML2.
